src/Voron/StorageEnvironment.cs

```diff
@@ -954,6 +954,8 @@ internal void TransactionCompleted(LowLevelTransaction tx)
                 _writeTransactionRunning.Reset();
                 _transactionWriter.Release();
 
+                Journal.Applicator.OnTransactionCompleted();
```

Comment:
Do we need to call it even if the transaction was committed? We might set `_waitForJournalStateUpdateUnderTx` twice for the same tx then, right? What could be the implication of doing that.

Reply:
2 options:
- This will cause us to retry acquiring the write tx lock.
- Will do nothing if we didn't do a flush.

Reply:
After the change, I'm checking if `_waitForJournalStateUpdateUnderTx` was set, if it was I'm not setting `_onWriteTransactionCompleted`.

Reply:
Great